lib/osf-components/addon/components/registries/update-dropdown/component.ts

```diff
@@ -0,0 +1,131 @@
+
+import RegistrationModel from 'ember-osf-web/models/registration';
+import RevisionModel, { RevisionReviewStates } from 'ember-osf-web/models/revision';
+import CurrentUserService from 'ember-osf-web/services/current-user';
+import Toast from 'ember-toastr/services/toast';
+import captureException, { getApiErrorMessage } from 'ember-osf-web/utils/capture-exception';
+import Store from '@ember-data/store';
+import RouterService from '@ember/routing/router-service';
+import { taskFor } from 'ember-concurrency-ts';
+import { computed } from '@ember/object';
+import registration from 'ember-osf-web/mirage/factories/registration';
+
+interface Args {
+    registration: RegistrationModel;
+}
+
+type RevisionJustification = 'Adding Results' | 'Typo - Self' | 'Typo - Other' | 'Copy Edit';
+
+export default class UpdateDropdown extends Component<Args> {
+    @service currentUser!: CurrentUserService;
+    @service intl!: Intl;
+    @service store!: Store;
+    @service toast!: Toast;
+    @service router!: RouterService;
+
+    revisions?: QueryHasManyResult<RevisionModel>;
+    isPendingCurrentUserApproval?: boolean;
+    revisionJustification?: RevisionJustification;
+
+    constructor(owner: unknown, args: Args) {
+        super(owner, args);
+        taskFor(this.getRevisionList).perform();
+    }
+
+    @computed('args.registration.{userHasAdminPermission,revisionState}')
+    get shouldDisplayApproveDenyButtons() {
+        return this.args.registration.userHasAdminPermission
+        && this.args.registration.revisionState
+        && ![
+            RevisionReviewStates.RevisionInProgress,
+            RevisionReviewStates.Approved,
+        ].includes(this.args.registration.revisionState);
+    }
+
+    @computed('args.registration.revisionState')
+    get updateNotificationIcon() {
+        switch (this.args.registration.revisionState) {
+        case RevisionReviewStates.Approved:
+            return 'lock';
+        case RevisionReviewStates.RevisionInProgress:
+            return 'eye';
+        case RevisionReviewStates.RevisionPendingAdminApproval:
+        case RevisionReviewStates.RevisionPendingModeration:
+            return 'clock';
+        default:
+            return '';
+        }
+    }
+
+    @task
+    @waitFor
+    async getRevisionList() {
+
+        if (!registration){
+            const notRegistrationError = this.intl.t('registries.update_dropdown.not_a_registration_error');
+            return this.toast.error(notRegistrationError);
+        }
+        if (this.args.registration.revisions === null || this.args.registration.revisions === undefined) {
+            return {
+                placeholder: this.intl.t('registries.update_dropdown.no_revisions_error'),
+            };
+        }
+        if (this.args.registration.revisions) {
+            try {
+                const revisions = await this.args.registration.queryHasMany('revisions');
+                this.revisions = revisions.sort();
```

Comment:
I prefer having a front end assurance that the list will display correctly, but Futa mentioned it may not be necessary. I have removed it, thank you Yuhuai.